homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,123 @@
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_name = light.name.strip()
+        self._attr_should_poll = False
+        self._light = light
+        self._light.subscribe(self)
+        if self._light.type == "dimmer":
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        else:
+            self._attr_color_mode = ColorMode.ONOFF
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, light.identifier)},
+            name=self._attr_name,
+            suggested_area=self._light.room,
+            via_device=(DOMAIN, api.mac),
+        )
+        self.update()
+        _LOGGER.debug("Initialising %s in %s", self.name, self._light.room)
+
+    @property
+    def available(self) -> bool:
+        """Return True if Home Assistant is able to read the state and control the underlying device."""
+        return self._light.is_connected
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+
+        if self._light.type == "dimmer":
+            return self._light.brightness * 255 / 100
+
+        return None
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light is on."""
+        return self._light.is_on
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Instruct the light to turn on (with optional brightness)."""
+
+        _LOGGER.debug(
+            "Sending turn_on(brightness=%d) for %s",
+            kwargs.get(ATTR_BRIGHTNESS, 255) * 100 / 255,
+            self.name,
+        )
+
+        if self._light.type == "dimmer":
+            await self._light.set_brightness(
+                kwargs.get(ATTR_BRIGHTNESS, 255) * 100 / 255
+            )
+        else:
+            await self._light.turn_on()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Instruct the light to turn off."""
+
+        _LOGGER.debug("Sending turn_off() for %s", self.name)
+
+        await self._light.turn_off()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cleanup ZimiLight with removal of notification prior to removal."""
+        await super().async_will_remove_from_hass()
+        self._light.unsubscribe(self)
+
+    def notify(self, _observable):
+        """Receive notification from light device that state has changed."""
+
+        _LOGGER.debug("Received notification() for %s", self.name)
+        self.schedule_update_ha_state(force_refresh=True)
+
+    def update(self) -> None:
+        """Fetch new state data for this light.
+
+        (The only data that needs updating is the name that might be set from Zimi app).
+        """
+
+        self._attr_name = self._light.name.strip()
```

Comment:
This is probably also pushed right? So we can just set it during your `notify`?

Reply:
I have moved to notify but I can't remove the update as it gets called (presumably because of self._attr_should_poll = False

Reply:
See https://github.com/home-assistant/core/pull/129876/commits/250c56e696d919814d0ce494e76e0c904313aa54